pkg/cmd/release/create/create.go

```diff
@@ -171,53 +177,238 @@ func createRun(cmd *cobra.Command, f factory.Factory) error {
 	return nil
 }
 
-func AskQuestions(octopus *octopusApiClient.Client, asker question.Asker, options *executor.TaskOptionsCreateRelease) error {
+func AskQuestions(octopus *octopusApiClient.Client, asker question.Asker, spinner factory.Spinner, options *executor.TaskOptionsCreateRelease) error {
+	if octopus == nil {
+		return errors.New("api client is required")
+	}
+	if asker == nil {
+		return errors.New("asker is required")
+	}
+	if options == nil {
+		return errors.New("options is required")
+	}
+	// Note: we don't get here at all if no-prompt is enabled, so we know we are free to ask questions
+
 	// Note on output: survey prints things; if the option is specified already from the command line,
 	// we should emulate that so there is always a line where you can see what the item was when specified on the command line,
 	// however if we support a "quiet mode" then we shouldn't emit those
 
 	var err error
 	var selectedProject *projects.Project
 	if options.ProjectName == "" {
-		selectedProject, err = selectProject(octopus, asker)
+		selectedProject, err = selectProject(octopus, asker, spinner)
 		if err != nil {
 			return err
 		}
 	} else { // project name is already provided, fetch the object because it's needed for further questions
-		selectedProject, err = findProject(octopus, options.ProjectName)
+		selectedProject, err = findProject(octopus, spinner, options.ProjectName)
 		if err != nil {
 			return err
 		}
 		_, _ = fmt.Printf("Project %s\n", output.Cyan(selectedProject.Name))
 	}
 	options.ProjectName = selectedProject.Name
 
-	if selectedProject.PersistenceSettings.GetType() == "VersionControlled" && options.GitReference != "" || options.GitCommit != "" {
-		// it's a config-as-code project, we need to ask for Git Ref or Commit
+	// we always need the deployment process, so we can prompt for package version overrides (or know that there aren't any packages, so it doesn't matter)
+
+	var gitReferenceKey string
+	if selectedProject.PersistenceSettings.GetType() == "VersionControlled" {
+		// if there is no git reference specified, ask the server for the list and prompt.
+		// we leave GitCommit alone in interactive mode; we don't prompt, but if it was specified on the
+		// commandline we just pass it through untouched.
+
+		if options.GitReference == "" { // we need a git ref; ask for one
+			gitRef, err := selectGitReference(octopus, asker, spinner, selectedProject)
+			if err != nil {
+				return err
+			}
+			options.GitReference = gitRef.Name // Hold the short name, not the canonical name due to golang url parsing bug replacing %2f with /
```

Comment:
🤮 but we've got to do it unless we make a breaking change to the api client to get rid of sling